src/AutorestSwift/View Models/OperationViewModel.swift

```diff
@@ -145,7 +153,24 @@ struct OperationViewModel {
         self.requiredQueryParams = requiredQueryParams
         self.requiredHeaders = requiredHeaders
         self.uriParams = uriParams
+        self.clientMethodOptionsViewModel = ClientMethodOptionsViewModel(
+            from: operation,
+            with: model,
+            parameters: optionsParams
+        )
+
+        for param in optionsParams {
+            assert(param.required == false, "Unexpectedly found a required 'option'... \(param.name)")
+        }
+    }
+}
+
+private func filterParams(for params: [Parameter], with allowed: [ParameterLocation]) -> [Parameter] {
+    let optionsParams = params.filter { param in
+        guard let httpParam = param.protocol.http as? HttpParameter else { return false }
+        return allowed.contains(httpParam.in)
     }
+    return optionsParams
 }
```

Comment:
```suggestion
private func filterParams(for params: [Parameter]?, with allowed: [ParameterLocation]) -> [Parameter] {
    guard params != nil else { return [Parameter]() }
    let optionsParams = params.filter { param in
        guard let httpParam = param.protocol.http as? HttpParameter else { return false }
        return allowed.contains(httpParam.in)
    }
    return optionsParams
}
```

Will let you simplify the invocations by not needed `?? []`